components/coding/profileV2/AchievementComponent.tsx

```diff
@@ -1,172 +1,161 @@
+} from "../../../graphql/coding/userBadges/updateUserCodingBadges";
+import UnitBadgeSection from "./achievementcomponenets/UnitBadgeSection";
+import handleOnSaveButtonClick from "./achievementcomponenets/handleSaveOnClick";
 
-export type BadgesSectionProps = {
-  data: User;
+export type AchievementComponentProps = {
+  userId: string;
+  isEditable: boolean;
 };
-export type userCodingBadge = {
-  id: number;
-};
-export type unitProps = {
-  unitTitle: string;
-  codingBadges: codingBadges[];
-};
-export type codingBadges = {
-  id: number;
-  title: string;
-  userCodingBadge: userCodingBadge[];
-  isAwarded: boolean;
-};
-export type unit = unitProps[];
-const AcheivementComponent = ({ user, isEditable = false }) => {
-  const { data } = useQuery(FETCH_CODING_BADGES, {
+
+const AcheivementComponent = ({
+  userId,
+  isEditable,
+}: AchievementComponentProps) => {
+  const [unitBadges, setUnitBadges] = useState<IntroCourseUnit[]>();
+  const [editMode, setEditMode] = useState(false);
+  const { data } = useQuery<FetchBadgeResponse>(FETCH_CODING_BADGES, {
+
     variables: {
-      userId: user.uid,
+      userId: userId,
+    },
+    onCompleted: (data) => {
+      setUnitBadges(data.intro_course_unit);
     },
   });
-  const [transformedData, setTransformedData] = useState([]);
-  const [editMode, setEditMode] = useState(false);
-  const handleBadgeClick = (inputBadge: codingBadges) => {
-    setTransformedData((prev) => {
-      const updatedData = prev.map((unit: unitProps) => {
-        return {
-          ...unit,
-          codingBadges: unit.codingBadges.map((badge) => {
-            if (badge.id == inputBadge.id) {
-              return {
-                ...badge,
-                isAwarded: !badge.isAwarded,
-              };
-            } else {
-              return badge;
-            }
-          }),
-        };
-      });
 
-      return updatedData;
+  const [saveAddedBadges] = useMutation(INSERT_USER_CODING_BADGES, {
+    refetchQueries: [{ query: FETCH_CODING_BADGES }],
+  });
+  const [saveRemovedBadges] = useMutation(DELETE_USER_CODING_BADGES, {
+    refetchQueries: [{ query: FETCH_CODING_BADGES }],
+  });
+
+  const separateBadges = (
+    changedBadgesList: CodingBadge[],
+    initialBadgeList: CodingBadge[]
+  ) => {
+    const addedBadgesTemp: CodingBadge[] = [];
+    const removedBadgesTemp: CodingBadge[] = [];
+    changedBadgesList.forEach((badge: CodingBadge) => {
+      const index = initialBadgeList.findIndex(
+        (initialBadge) => initialBadge.id === badge.id
+      );
+      if (
+        badge.user_coding_badges.length > 0 &&
+        initialBadgeList.find((initBadge) => badge.id == initBadge.id)
+          .user_coding_badges.length === 0
+      ) {
+        addedBadgesTemp.push(badge);
+      } else if (
+        badge.user_coding_badges.length === 0 &&
+        initialBadgeList[index].user_coding_badges.length > 0
+      ) {
+        removedBadgesTemp.push(badge);
+      }
+    });
```

Comment:
This function takes the changedBadgesList and initialBadgeList from findBadgeDiff and creates a list of addedBadges and removedBadges to send to the graphql mutation